DigitalExcellence/dex-backend
# Project Retrieval endpoints return Likes

## Description

Likes were not being returned when retrieving projects. Due to performance reasons, we are just returning an array for now with user id + date instead of Like Count and boolean to check if the currently logged-in user has liked that specific project. 
In the future, we should probably move the like and follow things into their own controller with respective services. The frontend can then make a call to retrieve project metadata, which can take a little longer and should be non-blocking for the frontend. 
For now, the frontend can count the objects in the likes array and check if a user-id is equal to the one that is logged in.
Fixes #329.

#335 should be merged before this as I merged that branch in this one to investigate performance while retrieving likes.

## Type of change


## Checklist


## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Make a call to project overview.
2. Check if object returns likes (make sure you add likes into DB)

## Link to issue

Closes: #329
